src/GafferSceneModule/EditScopeAlgoBinding.cpp

```diff
@@ -114,6 +114,23 @@ Imath::M44f matrixWrapper( EditScopeAlgo::TransformEdit &e )
 	return e.matrix();
 }
 
+bool hasParameterEditWrapper( const Gaffer::EditScope &scope, const ScenePlug::ScenePath &path, const std::string &attribute, const IECoreScene::ShaderNetwork::Parameter &parameter )
+{
+	return EditScopeAlgo::hasParameterEdit( &scope, path, attribute, parameter );
```

Comment:
No `ScopedGILRelease` here as `acquireParameterEdit( createIfNecessary = false)` only does c++ `GraphComponent` operations?

Reply:
It's not so much about C++, because signals and computes give us myriad opportunities to re-enter Python even when we think we're using C++. It's more about mutability - `hasParameterEdit()` doesn't change the graph, so won't trigger any signals. And it doesn't perform any computes, so that's another avenue to doom safely barricaded.